lib/cdo/crowdin/legacy_utils.rb

```diff
@@ -40,106 +40,84 @@ def initialize(project, options={})
       @logger = options.fetch(:logger, Logger.new(STDOUT))
     end
 
-    # Fetch from Crowdin a list of files changed since the last download. Uses
-    # etags sourced from the @etags_json file to define what we mean by "since
-    # the last download," and writes the results out to @files_to_download_json.
-    def fetch_changes
+    # Fetch from Crowdin a list of all of the project's files. Uses
+    # etags sourced from the @etags_json file to define what we mean by "changed",
+    # and updates the etags by writing the results out to @etags_json.
+    def download_changed_files
       etags = File.exist?(@etags_json) ? JSON.parse(File.read(@etags_json)) : {}
-      files_to_download = {}
+      # Initialize @files_to_sync_out file if it doesn't exist yet.
+      # It could already exist if multiple sync-down's occurred since the last sync-out.
+      File.write @files_to_sync_out_json, JSON.pretty_generate({}) unless File.exist?(@files_to_sync_out_json)
+      files_to_sync_out = JSON.parse(File.read(@files_to_sync_out_json))
+
+      files = @project.list_files
       languages = @project.languages
       num_languages = languages.length
+
       languages.each_with_index do |language, i|
         language_code = language["id"]
         @logger.debug("#{language['name']} (#{language_code}): #{i}/#{num_languages}")
         @logger.info("~#{(i * 100 / num_languages).round(-1)}% complete (#{i}/#{num_languages})") if i > 0 && i % (num_languages / 5) == 0
 
         etags[language_code] ||= {}
-        files = @project.list_files
+        # construct download directory; locale_subdir is optional, so compact
+        locale_dir = File.join([@locales_dir, language["name"], @locale_subdir].compact)
 
-        changed_files = Parallel.map(files, in_threads: MAX_THREADS) do |file|
+        downloaded_files = Parallel.map(files, in_threads: MAX_THREADS) do |file|
```

Comment:
I think it would be nice to have a rescue around this loop which saves the progress of the files we downloaded so far before an unrecoverable error was recovered.

Reply:
Good idea. Needing a sync complete by October 4th so I am going to say we create a follow up ticket for this specifically